pkg/applicationserver/applicationserver.go

```diff
@@ -401,21 +403,30 @@ func (as *ApplicationServer) DownlinkQueueList(ctx context.Context, ids ttnpb.En
 var errJSUnavailable = errors.DefineUnavailable("join_server_unavailable", "Join Server unavailable for JoinEUI `{join_eui}`")
 
 func (as *ApplicationServer) fetchAppSKey(ctx context.Context, ids ttnpb.EndDeviceIdentifiers, sessionKeyID []byte) (ttnpb.KeyEnvelope, error) {
-	// TODO: Lookup Join Server (https://github.com/TheThingsNetwork/lorawan-stack/issues/4)
-	js := as.GetPeer(ctx, ttnpb.PeerInfo_JOIN_SERVER, ids)
-	if js == nil {
-		return ttnpb.KeyEnvelope{}, errJSUnavailable.WithAttributes("join_eui", *ids.JoinEUI)
-	}
-	client := ttnpb.NewAsJsClient(js.Conn())
 	req := &ttnpb.SessionKeyRequest{
 		SessionKeyID: sessionKeyID,
 		DevEUI:       *ids.DevEUI,
 	}
-	res, err := client.GetAppSKey(ctx, req, as.WithClusterAuth())
-	if err != nil {
-		return ttnpb.KeyEnvelope{}, err
+	if js := as.GetPeer(ctx, ttnpb.PeerInfo_JOIN_SERVER, ids); js != nil {
+		res, err := ttnpb.NewAsJsClient(js.Conn()).GetAppSKey(ctx, req, as.WithClusterAuth())
+		if err == nil {
+			return res.AppSKey, nil
+		}
+		if !errors.IsNotFound(err) {
+			return ttnpb.KeyEnvelope{}, err
+		}
+	}
+	if as.interopClient != nil {
+		// TODO: Set AS-ID.
```

Comment:
Not sure what to put here - @htdvisser @johanstokking ?

Reply:
AFAIK the AS-ID is just the hostname of the AS, but let's wait for @johanstokking

Reply:
I think this should come from config, I'm afraid.

Yes, TTN JS uses the host of `application_server_address` as AS-ID.

Other (non-TTN) JS issue AS IDs, and they may contain characters that are not valid in hosts, i.e. characters that do not pass the regex we have on our `*_address` fields. Still, TTN AS should identify itself with that AS ID, and since it can be something arbitrary from a JS, I think we should add this to AS config.